Runtime/Scripts/LocationService.cs

```diff
@@ -99,27 +93,39 @@ public static IEnumerator InitalizeLocationService(int maxWait = 20, bool contin
 
         /// <summary>
         /// EnsureLocation Confirm that user location is valid, user location is essential for MobiledgeX services
-        /// If Location permission is denied by User an exception will be thrown (for Android) in LocationServiceFlow()
-        ///                                                                      (for iOS) in InitalizeLocationService()
+        /// If Location permission is denied by User an exception will be thrown once RetrieveLocation() is called
         /// </summary>
         public static IEnumerator EnsureLocation()
         {
+            if (!SystemInfo.supportsLocationService)
+            {
+                yield return null;
+            }
+
 #if UNITY_EDITOR
             yield return null;
 #else
             if (Input.location.status == LocationServiceStatus.Initializing)
             {
-                yield return new WaitUntil(() => (Input.location.status == LocationServiceStatus.Running));
+                yield return new WaitUntil(() => (Input.location.status != LocationServiceStatus.Initializing));
+            }
+            if(locationPermissionRejected == true || Input.location.status == LocationServiceStatus.Failed || !Input.location.isEnabledByUser)
```

Comment:
nit: space between "if" and "("

Reply:
Also, shouldn't the locationPermissionRejected check happen before we check the Input.location.status?